pkg/kubeinteraction/status/task_status.go

```diff
@@ -113,8 +115,13 @@ func CollectFailedTasksLogSnippet(ctx context.Context, cs *params.Run, kinteract
 					if strings.HasSuffix(trimmed, " Skipping step because a previous step failed") {
 						continue
 					}
-					// see if a pattern match from errRe
-					ti.LogSnippet = strings.TrimSpace(trimmed)
+					if len(trimmed) > maxErrorSnippetCharacterLimit {
+						runes := []rune(trimmed)
+						if len(runes) > maxErrorSnippetCharacterLimit {
+							trimmed = string(runes[:maxErrorSnippetCharacterLimit])
+						}
+					}
+					ti.LogSnippet = trimmed
```

Comment:
This can be made a bit cleaner (and technically more performant, though that might not matter much here) with `unicode/utf8`'s `RunesInString` function:
```suggestion
					if len(trimmed) > maxErrorSnippetCharacterLimit && utf8.RunesInString(trimmed) > maxErrorSnippetCharacterLimit {
						runes := []rune(trimmed)
						trimmed = string(runes[:maxErrorSnippetCharacterLimit])
						}
					}
					ti.LogSnippet = trimmed
```

Reply:
done, thanks for the hint